crates/orderbook/src/account_balances.rs

```diff
@@ -25,10 +25,18 @@ impl Query {
     }
 }
 
+#[derive(Debug, PartialEq, Eq)]
+pub enum TransferSimulationResult {
+    Ok,
+    InsufficientAllowance,
+    InsufficientBalance,
+    TransferFailed,
+}
```

Comment:
ubernit: Not a fan of these `CustomResult` types because they miss out on all the `Result` goodies (`?` support, all the standard library methods like `map`, `unwrap_or_else`, etc.) - I would much prefer:
```suggestion
pub enum TransferSimulationError {
    InsufficientAllowance,
    InsufficientBalance,
    TransferFailed,
    Other(anyhow::Error),
}
```

And then return `Result<(), TransferSimulationError>`.

In terms of how to go from `anyhow::Error` to `TransferSimulationError`, you can either
```rust
impl From<anyhow::Error> for TransferSimulationError {
    fn from(err: anyhow::Error) -> Self {
        Self::Other(err)
    }
}
```

Or, you can manually do it at the call site:
```rust
erc20_balance_query(&mut batch, token, from, self.vault_relayer)
    .await
    .map_err(TransferSimulationError::Other)?;
```